core/meterer/meterer.go

```diff
@@ -41,13 +44,248 @@ func NewMeterer(
 	offchainStore OffchainStore,
 	logger logging.Logger,
 ) (*Meterer, error) {
-	// TODO: create a separate thread to pull from the chain and update chain state
-	return &Meterer{
+	ctx := context.Background()
+	m := Meterer{
 		Config: config,
 
 		ChainState:    paymentChainState,
 		OffchainStore: offchainStore,
 
 		logger: logger.With("component", "Meterer"),
-	}, nil
+	}
+	// goroutine to periodically refresh the on-chain state
+	go func() {
```

Comment:
Can we have `Start()` method instead of starting a new goroutine in constructor?